low-level/framework/messaging/thread_controller.py

```diff
@@ -48,38 +51,159 @@
+    # Suspend module threads
+    if resume == False:
+        module.suspend()
+
+    # Initialize persistent cache for sensor status
+    per_data_path = os.path.join(
+        module_cache_dir, f"{module_name.upper()}_{node_id}")
+    if not os.path.isfile(per_data_path):
+        module_persistent_data[module_name] = {}
+        store.put(module_persistent_data[module_name], per_data_path)
+
+    is_sensor_thread = False
+    recovery_count = recovery_interval = 0
+    if isinstance(module, SensorThread):
+        recovery_count, recovery_interval = _get_recovery_config(module_name)
+        is_sensor_thread = True
+
+    attempt = 0
+
+    while attempt <= recovery_count:
+        attempt += 1
+        try:
+            # Each module is passed a reference list to message queues so it
+            # can transmit internal messages to other modules as desired
+            module.start_thread(conf_reader, msgQlist, product)
+        except Exception as err:
+            curr_state = "fault"
+            err_msg = f"{module_name}, {err}"
+            logger.error(err_msg)
+            if attempt > recovery_count:
+                logger.debug(traceback.format_exc())
+                description = f"{module_name} is stopped and unrecoverable. {err_msg}"
+                impact = module.impact()
+                recommendation = "Restart SSPL service"
+                logger.critical(
+                    f"{description}. Impact: {impact} Recommendation: {recommendation}")
+                # Check previous state of the module and send fault alert
+                if os.path.isfile(per_data_path):
+                    module_persistent_data[module_name] = store.get(per_data_path)
+                prev_state = module_persistent_data[module_name].get('prev_state')
+                if is_sensor_thread and curr_state != prev_state:
+                    module_persistent_data[module_name] = {"prev_state": curr_state}
+                    store.put(module_persistent_data[module_name], per_data_path)
+                    specific_info = Conf.get(SSPL_CONF, f"{module_name.upper()}")
+                    info = {
+                        "module_name": module_name,
+                        "alert_type": curr_state,
+                        "description": description,
+                        "impact": impact,
+                        "recommendation": recommendation,
+                        "severity": "critical",
+                        "specific_info": specific_info
+                    }
+                    jsonMsg = ThreadMonitorMsg(info).getJson()
+                    module._write_internal_msgQ(EgressProcessor.name(), jsonMsg)
+            else:
+                logger.debug(f"Recovering {module_name} from failure, "
+                             f"attempt: {attempt}")
+                time.sleep(recovery_interval)
+
+            # Shutdown if no recovery attempt
+            logger.info(f"Shutting down, {module_name}")
```

Comment:
Dont keep comma in between, it gives feeling that SSPL is shutting down. 
> Terminating monitoring thread  {module_name}

Reply:
Corrected,
Done.